x-pack/plugins/actions/server/plugin.ts

```diff
@@ -376,6 +375,9 @@ export class ActionsPlugin implements Plugin<PluginSetupContract, PluginStartCon
         };
       },
       getActionsConfigurationUtilities: () => actionsConfigUtils,
+      setEnabledConnectorTypes: (actionTypes) => {
+        actionTypeRegistry.setEnabledConnectorTypes(new Set(actionTypes));
```

Comment:
Should we allow consumers of the API to disable system action types?

Reply:
This is a good question, as we matched the EnabledConnectorTypes list in the config if the connectorType is not in the enabled list (both registry and config) it wouldn't be executed.
IMO we should always allow them.

Sounds like a bug and needs to be fixed in:

https://github.com/elastic/kibana/blob/main/x-pack/plugins/actions/server/action_type_registry.ts#L71-L79

Should we allow system actions always to be executed, even if the connector type is not in the enabled lists (config, license, registry)? if so i can fix it in this PR.

cc: @mikecote

Reply:
If I remember correctly we decided to allow administrators to disable system actions using `kibana.yml`. This is probably why the test is there. Regarding licensing, they should not be able to use the system action if they do not have the proper license. The case action would probably be available only in Platinum+. Given that, should we allow other developers to disable system action using the new API functions? I am not sure 🙂. Probably yes. Just wanted to start a conversation.

Reply:
>Given that, should we allow other developers to disable system action using the new API functions? I am not sure 🙂. Probably yes. Just wanted to start a conversation.

I think whatever happens when setting system actions in `xpack.actions.enabledActionTypes` should be reflected within `setEnabledConnectorTypes(...)` for consistency. This feature should be as close as `xpack.actions.enabledActionTypes` but allows plugin developers to set it dynamically during setup instead.

Reply:
Make sense to me!